platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/internal/AbstractTaskSchedulerBuilder.java

```diff
@@ -130,12 +130,31 @@ public AbstractTaskSchedulerBuilder<OUT> configure(@NonNull final TaskSchedulerC
         return this;
     }
 
+    /**
+     * Ensures that direct schedulers do not have an unhandled task capacity set.
+     *
+     * <p>If the scheduler type is {@link TaskSchedulerType#DIRECT} or {@link TaskSchedulerType#DIRECT_THREADSAFE}
+     * and the unhandled task capacity is not 1, an {@link IllegalArgumentException} is thrown.
+     *
+     * @param type the type of the task scheduler
+     * @param unhandledTaskCapacity the unhandled task capacity
+     * @throws IllegalArgumentException if the type is direct or direct threadsafe and the unhandled task capacity is not 1
+     */
+    public static void validateDirectScheduler(
+            @NonNull final TaskSchedulerType type, final long unhandledTaskCapacity) {
+        if ((type == TaskSchedulerType.DIRECT || type == TaskSchedulerType.DIRECT_THREADSAFE)
+                && unhandledTaskCapacity != 1) {
+            throw new IllegalArgumentException("Direct schedulers cannot have an unhandled task capacity.");
+        }
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     @NonNull
     public AbstractTaskSchedulerBuilder<OUT> withType(@NonNull final TaskSchedulerType type) {
+        validateDirectScheduler(type, this.unhandledTaskCapacity);
```

Comment:
I'd suggest you only do this check when the scheduler is built. There could be some weirdness where it is possible to get an error while entering a legal combination of configurations but in the wrong order.